generators/commons/src/utils/parseIR.ts

```diff
@@ -0,0 +1,82 @@
+import { AbsoluteFilePath, streamObjectFromFile } from "@fern-api/fs-utils";
+
+export declare namespace parseIR {
+    export interface Args<IntermediateRepresentation> {
+        absolutePathToIR: AbsoluteFilePath;
+        parse: (raw: unknown, opts?: SchemaOptions) => MaybePromise<MaybeValid<IntermediateRepresentation>>;
+    }
+
+    /* Copy some types across the IR SDKs */
+    export type MaybePromise<T> = T | Promise<T>;
+
+    export type MaybeValid<T> = Valid<T> | Invalid;
+
+    export interface Valid<T> {
+        ok: true;
+        value: T;
+    }
+
+    export interface Invalid {
+        ok: false;
+        errors: unknown[];
+    }
+
+    interface SchemaOptions {
+        /**
+         * how to handle unrecognized keys in objects
+         *
+         * @default "fail"
+         */
+        unrecognizedObjectKeys?: "fail" | "passthrough" | "strip";
+        /**
+         * whether to fail when an unrecognized discriminant value is
+         * encountered in a union
+         *
+         * @default false
+         */
+        allowUnrecognizedUnionMembers?: boolean;
+        /**
+         * whether to fail when an unrecognized enum value is encountered
+         *
+         * @default false
+         */
+        allowUnrecognizedEnumValues?: boolean;
+        /**
+         * whether to allow data that doesn't conform to the schema.
+         * invalid data is passed through without transformation.
+         *
+         * when this is enabled, .parse() and .json() will always
+         * return `ok: true`. `.parseOrThrow()` and `.jsonOrThrow()`
+         * will never fail.
+         *
+         * @default false
+         */
+        skipValidation?: boolean;
+        /**
+         * each validation failure contains a "path" property, which is
+         * the breadcrumbs to the offending node in the JSON. you can supply
+         * a prefix that is prepended to all the errors' paths. this can be
+         * helpful for zurg's internal debug logging.
+         */
+        breadcrumbsPrefix?: string[];
+    }
+}
+
+export async function parseIR<IR>({ absolutePathToIR, parse }: parseIR.Args<IR>): Promise<IR> {
+    const irJson = await streamObjectFromFile(absolutePathToIR);
+    // eslint-disable-next-line no-console
+    console.log(`Read ${absolutePathToIR}`);
```

Comment:
Thoughts on the following?

```suggestion
    console.log(`Parsing ${absolutePathToIR} ...`);
```

Otherwise, I feel like we'll forget what stage this got to before we receiving an error. It'll be _slightly_ more clear that we're about to parse the IR.

Totally fine with other options too, just feel like `Read` is not clear enough.

Reply:
Sounds great, I basically just kept in line with https://github.com/fern-api/fern/blob/5ec83da6ede3c1b6019c73db4b2e4b793ddf6fca/generators/commons/src/utils/parseGeneratorConfig.ts#L8 but down to change